### ISSUE TITLE
Sumtype and Interface Leaks

### ISSUE THREAD

**Issue by amoyyy** [vlang/v#12023]
**V version:** latest
**OS:** ubuntu 20.04

**What did you do?**
`test_leak.v`:
```
module main 

struct PrimitiveSolverA {}
struct PrimitiveSolverB {}

interface ISolver{}
type SSolver = PrimitiveSolverA | PrimitiveSolverB

fn new_solver_interface() ISolver{
	return PrimitiveSolverA{}
}

fn new_solver_sumtype() SSolver{
	return PrimitiveSolverB{}
}

fn main(){
	s1 := new_solver_interface()
	s2 := new_solver_sumtype()
}
```
`v -autofree test_leak.v && valgrind --leak-check=full ./test_leak`

**What did you expect to see?**
no memory leak.
 
**What did you see instead?**
The transversion from `struct` instance to `sumtype` or `interface` leads to memory leaks. And the more fields the struct has, the more allocations and leaks happen. It's a basic design pattern used by coders, will it be fixed or are there any ways to avoid the leaks? Grateful for advice.

relevant issues:#6456 Using sum types causes leaks

**Comment by dumblob** [comment]
`-autofree` is still unfinished, so thanks for reporting!

**Comment by medvednikov** [comment]
Fixed in 0.3.
